algebra/src/to_field_vec.rs

```diff
@@ -42,116 +41,41 @@ impl<ConstraintF: Field> ToConstraintField<ConstraintF> for () {
-{
-    fn to_field_elements(&self) -> Result<Vec<P::BasePrimeField>, Error> {
-        let mut res = Vec::new();
-        let mut c0_elems = self.c0.to_field_elements()?;
-        let mut c1_elems = self.c1.to_field_elements()?;
-
-        res.append(&mut c0_elems);
-        res.append(&mut c1_elems);
-
-        Ok(res)
-    }
-}
-
-impl<P: CubicExtParameters> ToConstraintField<P::BasePrimeField> for CubicExtField<P>
-where
-    P::BaseField: ToConstraintField<P::BasePrimeField>,
-{
-    fn to_field_elements(&self) -> Result<Vec<P::BasePrimeField>, Error> {
-        let mut res = Vec::new();
-        let mut c0_elems = self.c0.to_field_elements()?;
-        let mut c1_elems = self.c1.to_field_elements()?;
-        let mut c2_elems = self.c2.to_field_elements()?;
-
-        res.append(&mut c0_elems);
-        res.append(&mut c1_elems);
-        res.append(&mut c2_elems);
-
-        Ok(res)
-    }
-}
-
-impl<M: TEModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for TEAffine<M>
-where
-    M::BaseField: ToConstraintField<ConstraintF>,
+impl<M: SWModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for Jacobian<M>
+    where
+        M::BaseField: ToConstraintField<ConstraintF>,
 {
     #[inline]
     fn to_field_elements(&self) -> Result<Vec<ConstraintF>, Error> {
-        let mut x_fe = self.x.to_field_elements()?;
-        let y_fe = self.y.to_field_elements()?;
-        x_fe.extend_from_slice(&y_fe);
-        Ok(x_fe)
-    }
-}
-
-impl<M: TEModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for TEProjective<M>
-where
-    M::BaseField: ToConstraintField<ConstraintF>,
-{
-    #[inline]
-    fn to_field_elements(&self) -> Result<Vec<ConstraintF>, Error> {
-        let affine = self.into_affine();
+        let affine = self.into_affine()?;
         let mut x_fe = affine.x.to_field_elements()?;
         let y_fe = affine.y.to_field_elements()?;
         x_fe.extend_from_slice(&y_fe);
         Ok(x_fe)
     }
 }
 
-impl<M: SWModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for SWJAffine<M>
-where
-    M::BaseField: ToConstraintField<ConstraintF>,
-{
-    #[inline]
-    fn to_field_elements(&self) -> Result<Vec<ConstraintF>, Error> {
-        let mut x_fe = self.x.to_field_elements()?;
-        let y_fe = self.y.to_field_elements()?;
-        x_fe.extend_from_slice(&y_fe);
-        Ok(x_fe)
-    }
-}
-
-impl<M: SWModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for SWJProjective<M>
-where
-    M::BaseField: ToConstraintField<ConstraintF>,
+impl<M: SWModelParameters, ConstraintF: Field> ToConstraintField<ConstraintF> for Projective<M>
+    where
+        M::BaseField: ToConstraintField<ConstraintF>,
 {
     #[inline]
     fn to_field_elements(&self) -> Result<Vec<ConstraintF>, Error> {
-        let affine = self.into_affine();
+        let affine = self.into_affine()?; // Affine coordinates are defined even if `self` is the neutral elements
```

Comment:
As this was fixed later on, I removed the comment and put a TODO instead, that we will remove when the branch that fixed it will rebase on top of this